### PR TITLE
Phase 2.1: Add versioned WAL format and walStore

### PR DESCRIPTION
**Pull request by Ayan-sh03**
Implements issue #26: add the WAL foundation without switching handlers yet.\n\nCloses #26.\n\nChanges:\n- add versioned KWAL frame encoding/decoding with CRC validation\n- add WAL op and payload types for create, publish, claim, ack, nack, and reap batches\n- add Pebble-backed walStore.Append and walStore.Replay\n- add KUEUE_WAL_SYNC parsing for none, batch, and always\n- add focused WAL tests for round trips, corruption, ordering, replay, and sync parsing\n\nValidation:\n- go test ./...\n- go build ./...

## Summary by CodeRabbit

* **New Features**
  * Added write-ahead log functionality with configurable synchronization modes (`none`, `batch`, `always`).

* **Tests**
  * Added comprehensive test coverage for write-ahead log operations, encoding/decoding, and edge cases including corruption detection.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Ayan-sh03/kueue/pull/36?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: wal.go [hunk 1]
```diff
@@ -0,0 +1,1033 @@
...
+		}
+		entry, err := decodeWalEntry(lsn, val)
+		if err != nil {
+			return err
+		}
+		if err := apply(entry); err != nil {
+			return err
+		}
+	}
+	return nil
```
[the remark this answers is not in the document]
Comment (Ayan-sh03) [reply wal.go]: done
